src/js/webrequest.js

```diff
@@ -587,30 +588,40 @@ let getWidgetBlockList = (function () {
       translations.rtl = RTL_LOCALES.indexOf(UI_LOCALE) > -1;
     }
 
-    badger.widgetList.forEach(function (widget) {
-      let replace = false;
+    for (let i = 0; i < badger.widgetList.length; i++) {
```

Comment:
why move from a `forEach(...)` to a classic for loop?

Reply:
You can't short-circuit `forEach` loops.